qcodes/instrument_drivers/tektronix/Keithley_2450.py

```diff
@@ -1,11 +1,171 @@
+    }
+
+    def __init__(
+            self,
+            parent: 'Keithley2450',
+            name: str,
+            size: Optional[int] = None,
+            style: str = ''
+    ) -> None:
+        super().__init__(parent, name)
+        self.buffer_name = name
+        self._size = size
+        self.style = style
+
+        if self.buffer_name not in self.default_buffer:
+            # when making a new buffer, the "size" parameter is required.
+            if size is None:
+                raise TypeError(
+                    "buffer() missing 1 required positional argument: 'size'"
+                )
+            self.write(
+                f":TRACe:MAKE '{self.buffer_name}', {self._size}, {self.style}"
+            )
+        else:
+            # when referring to default buffer, "size" parameter is not needed.
+            if size is not None:
+                self.log.warning(
+                    f"Please use method 'size()' to resize default buffer "
+                    f"{self.buffer_name} size to {self._size}."
+                )
+
+        self.add_parameter(
+            "size",
+            get_cmd=f":TRACe:POINts? '{self.buffer_name}'",
+            set_cmd=f":TRACe:POINts {{}}, '{self.buffer_name}'",
+            get_parser=int,
+            docstring="The number of readings a buffer can store."
+        )
+
+        self.add_parameter(
+            "number_of_readings",
+            get_cmd=f":TRACe:ACTual? '{self.buffer_name}'",
+            get_parser=int,
+            docstring="To get the number of readings in the reading buffer."
+        )
+
+        self.add_parameter(
+            "elements",
+            get_cmd=None,
+            set_cmd=None,
+            vals=Lists(Enum('date', 'measurement_formatted',
+                            'fractional_seconds', 'measurement',
+                            'relative_time', 'seconds', 'source_value',
+                            'source_value_formatted', 'source_value_status',
+                            'source_value_unit', 'measurement_status', 'time',
+                            'timestamp', 'measurement_unit')),
+            docstring="List of buffer elements to read."
+        )
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.delete()
+
+    @property
+    def available_elements(self) -> set:
+        return set(self.buffer_elements.keys())
+
+    def get_last_reading(self) -> str:
+        """
+        This method requests the latest reading from a reading buffer.
+
+        """
+        if (self.elements() is None) or (len(self.elements()) == 0):
+            return self.ask(f":FETCh? '{self.buffer_name}'")
+        fetch_elements = [
+            self.buffer_elements[element] for element in self.elements()
+        ]
+        return self.ask(
+            f":FETCh? '{self.buffer_name}', {','.join(fetch_elements)}"
+        )
+
+    def get_data(self, start_idx: int, end_idx: int) -> list:
```

Comment:
I put all the ":TRACE" related cmd under Buffer class. Maybe I should rewrite the `_measure_sweep()` with this `get_data()` method.

Reply:
> Maybe I should rewrite the _measure_sweep() with this get_data() method.

that sounds like a good idea - removing duplication and properly using the buffer infra :)

Reply:
I added an extra argument to the `get_data` method, and rewrite the `_measure_sweep`. Now less code repetition, works the same.